### PR TITLE
chore: Only run E2E suite when ready-for-e2e label added

### PR DESCRIPTION
**Pull request by keithwillcode**
## What does this PR do?

Gives us optionality in terms of not always running the full E2E suite until manually triggered by adding the ‘ready-for-e2e’ label or when an approval happens, which automatically adds said label. We still protect the main branch because the required checks waits for the e2e test suite to complete and pass. 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?
- ensure that the PR update job run manually against this branch has succeeded.

### CODE REVIEW

File: .github/workflows/nextjs-bundle-analysis.yml [hunk 1]
```diff
@@ -45,13 +45,7 @@ env:
   TURBO_TEAM: ${{ secrets.TURBO_TEAM }}
 
 jobs:
-  build:
```
Comment (keithwillcode) [inline .github/workflows/nextjs-bundle-analysis.yml:48]: This is a duplicate call to run a build

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -27,37 +28,132 @@ jobs:
...
   integration-test:
     name: Tests
-    needs: [changes]
-    if: ${{ needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
+    needs: [changes, check-label]
+    if: ${{ needs.check-label.outputs.run-jobs == 'true' && needs.changes.outputs.has-files-requiring-all-checks == 'true' }}
     uses: ./.github/workflows/integration-tests.yml
     secrets: inherit
 
+  build-api-v1:
```
Comment (keithwillcode) [inline .github/workflows/pr.yml:99]: Moving all of the checks back into PR flow which enables us to disable the merge queue

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -27,37 +28,149 @@ jobs:
           filters: |
             has-files-requiring-all-checks:
               - "!(**.md|.github/CODEOWNERS)"
+
+  check-label:
+    runs-on: buildjet-2vcpu-ubuntu-2204
+    name: Check for E2E label
+    outputs:
+      run-e2e: ${{ steps.check-if-pr-has-label.outputs.run-e2e == 'true' && (github.event_name != 'labeled' || (github.event_name == 'labeled' && github.event.label.name == 'ready-for-e2e')) }}
+      run-jobs: ${{ github.event_name != 'labeled' }}
```
Comment (joeauyeung) [inline .github/workflows/pr.yml:37]: Do we want to add the conditional for the `unlabeled` event as well?
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label

Comment (keithwillcode) [reply .github/workflows/pr.yml:37]: No, because the workflow isn't triggered at all for unlabeled

Comment (joeauyeung) [reply .github/workflows/pr.yml:37]: Ah I see where now. This makes sense.

**Review by joeauyeung — APPROVED** [review]
LGTM

**Pull request merged**